File: src/main/java/com/mpe85/grampa/input/impl/StringBuilderInputBuffer.java
package com.mpe85.grampa.input.impl;

import com.mpe85.grampa.input.InputBuffer;

/**
 * An {@link InputBuffer} implementation using a {@link StringBuilder}.
 * 
 * @author mpe85
 *
 */
public class StringBuilderInputBuffer extends CharSequenceInputBuffer {
	
	private final StringBuilder stringBuilder;
	
	public StringBuilderInputBuffer(final StringBuilder stringBuilder) {
		super(stringBuilder);
		this.stringBuilder = stringBuilder;
	}
	
	@Override
	public int getCodePoint(final int index) {
		return stringBuilder.codePointAt(index);
	}
	
}

File: src/main/kotlin/com/mpe85/grampa/input/impl/StringBuilderInputBuffer.kt
package com.mpe85.grampa.input.impl

import com.mpe85.grampa.input.InputBuffer

/**
 * An [InputBuffer] implementation using a [StringBuilder].
 *
 * @author mpe85
 * @param[stringBuilder] A string builder backing the input buffer
 */
class StringBuilderInputBuffer(private val stringBuilder: StringBuilder) : CharSequenceInputBuffer(stringBuilder) {

    override fun getCodePoint(index: Int) = stringBuilder.codePointAt(index)

}
